src/forwarder.ts

```diff
@@ -142,15 +142,22 @@ export async function addStepFunctionLogGroup(aws: Aws, resources: any, stepFunc
 }
 
 export function addDdSlsPluginTag(stateMachineObj: any) {
-  if (stateMachineObj && stateMachineObj.Properties && !stateMachineObj.Properties.Tags) {
-    stateMachineObj.Properties.Tags = [];
-  }
   stateMachineObj.Properties?.Tags?.push({
     Key: "dd_sls_plugin",
     Value: `v${version}`,
   });
 }
 
+export function addDdTraceEnabledTag(stateMachineObj: any, enableStepFunctionsTrace: undefined | boolean) {
+  if (!enableStepFunctionsTrace) {
```

Comment:
So the logic is: Unless the user specifically set enableStepTunctionsTrace to be false, DD_Trace_Enabled will always be set true, and step function tracing is enabled. Right?

Reply:
I think I missed the logic where we set this flag default to true. And I'll need to catch all the possible exceptions for this logic in case there are any errors that may block any customers' deployment.

Reply:
I just realized that we cannot set this flag to true. So there is no need to add try catch for it then.